src/structure/model.jl

```diff
@@ -663,463 +956,658 @@ Base.eltype(::Type{Residue}) = AbstractAtom
-waterselector(element::Union{AbstractResidue, AbstractAtom}) = resnameselector(element, water_res_names)
+function waterselector(el::Union{AbstractResidue, AbstractAtom})
+    return resnameselector(el, waterresnames)
+end
+
 
 """
 Determines if an `AbstractResidue` represents a standard protein residue,
-i.e. consists of ATOM records.
+e.g. consists of ATOM records in a Protein Data Bank (PDB) file.
 """
 stdresselector(res::AbstractResidue) = !ishetres(res)
 
+
 """
-Determines if an `AbstractResidue` represents a hetero molecule, i.e. consists
-of HETATM records.
+Determines if an `AbstractResidue` represents a hetero molecule, e.g. consists
+of HETATM records in a Protein Data Bank (PDB) file.
 """
 hetresselector(res::AbstractResidue) = ishetres(res)
 
+
 """
 Determines whether an `AbstractAtom` or `AbstractResidue` is disordered, i.e.
 has multiple locations in the case of atoms or multiple residue names (point
 mutants) in the case of residues.
 """
-disorderselector(atom::AbstractAtom) = isdisorderedatom(atom)
+disorderselector(at::AbstractAtom) = isdisorderedatom(at)
+
 disorderselector(res::AbstractResidue) = isdisorderedres(res)
 
+
 # Either the element is H or the element field is empty, the atom name contains
 #   an H and there are no letters in the atom name before H
-# For example atom names 1H and H1 would be hydrogens but NH1 would not
+# For example atom names "1H" and "H1" would be hydrogens but "NH1" would not
 """
 Determines if an `AbstractAtom` represents hydrogen. Uses the element field
 where possible, otherwise uses the atom name.
 """
-hydrogenselector(atom::AbstractAtom) = element(atom) == "H" || (element(atom) == "" && 'H' in atomname(atom) && !ismatch(r"[a-zA-Z]", atomname(atom)[1:findfirst(atomname(atom), 'H')-1]))
+function hydrogenselector(at::AbstractAtom)
+    return element(at, spaces=false) == "H" ||
+        (element(at, spaces=false) == "" &&
+        'H' in atomname(at) &&
+        !ismatch(r"[a-zA-Z]", atomname(at)[1:findfirst(atomname(at), 'H')-1]))
+end
+
+
+# Residues are ordered in a chain with all hetero residues at the end
+# For obtaining sequence we will re-order them numerically
+function AminoAcidSequence(ch::Chain, residue_selectors::Function...)
+    return AminoAcidSequence(
+        sort(collectresidues(ch, residue_selectors...), by=resnumber))
+end
+
+function AminoAcidSequence{T <: AbstractResidue}(res::Vector{T})
+    seq = AminoAcid[]
+    for i in 1:length(res)
+        if resname(res[i]) in keys(threeletter_to_aa)
+            push!(seq, threeletter_to_aa[resname(res[i])])
+        else
+            push!(seq, AA_X)
+        end
+        if i+1 <= length(res) && resnumber(res[i+1]) - resnumber(res[i]) > 1
+            append!(seq, [AA_Gap for _ in 1:(resnumber(res[i+1]) - resnumber(res[i]) - 1)])
+        end
+    end
+    return AminoAcidSequence(seq)
+end
 
 
 # Descriptive showing of elements
 
 function Base.show(io::IO, struc::ProteinStructure)
     if countmodels(struc) > 0
```

Comment:
Showing the type name first would be nice (e.g. `println(io, summary(struct))`).